Emby.Server.Implementations/Data/SqliteItemRepository.cs

```diff
@@ -6130,5 +6144,170 @@ private MediaStream GetMediaStream(IReadOnlyList<IResultSetValue> reader)
 
             return item;
         }
+
+        public List<MediaAttachment> GetMediaAttachments(MediaAttachmentQuery query)
+        {
+            CheckDisposed();
+
+            if (query == null)
+            {
+                throw new ArgumentNullException(nameof(query));
+            }
+
+            var cmdText = "select "
+                        + string.Join(",", _mediaAttachmentSaveColumns)
+                        + " from mediaattachments where"
+                        + " ItemId=@ItemId";
+
+            if (query.Index.HasValue)
+            {
+                cmdText += "AND AttachmentIndex=@AttachmentIndex";
+            }
+
+            cmdText += " order by AttachmentIndex ASC";
+
+            using (var connection = GetConnection(true))
+            {
+                var list = new List<MediaAttachment>();
+
+                using (var statement = PrepareStatement(connection, cmdText))
+                {
+                    statement.TryBind("@ItemId", query.ItemId.ToGuidBlob());
+
+                    if (query.Index.HasValue)
+                    {
+                        statement.TryBind("@AttachmentIndex", query.Index.Value);
+                    }
+
+                    foreach (var row in statement.ExecuteQuery()) {
+                        list.Add(GetMediaAttachment(row));
+                    }
+                }
+
+                return list;
+            }
+        }
+
+        public void SaveMediaAttachments(Guid id, List<MediaAttachment> attachments, CancellationToken cancellationToken)
```

Comment:
why `cancellationToken` is neither used nor passed further down to `InsertMediaAttachments`?